src/marqo/s2_inference/hf_utils.py

```diff
@@ -75,10 +164,46 @@ def forward(self, **kwargs):
     def mean_pooling(self, model_output, attention_mask):
 
         token_embeddings = model_output[0]
-        
+
         input_mask_expanded = attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
 
         return torch.sum(token_embeddings * input_mask_expanded, 1) / torch.clamp(input_mask_expanded.sum(1), min=1e-9)
 
     def cls_pooling(self, model_output, attention_mask):
-        return model_output[0][:,0]
+        return model_output[0][:, 0]
+
+
+def validate_huggingface_archive(path: str) -> str:
+    '''
+        This function extracts the model from a huggingface archive if necessary.
+        Unlike open clip models that can loaded from a single .pt or .bin file,
+        Huggingface models are loaded from a directory.
+
+    Returns:
+        The directory path to the model
+    '''
+    if os.path.isdir(path):
+        # if it's a directory, return the path directly
+        return path
+    elif os.path.isfile(path):
+        # if it's a file, check if it's a compressed file
+        base, ext = os.path.splitext(path)
+        if ext in ['.tar', '.gz', '.tgz', '.bz2', '.zip']:
```

Comment:
It may be worth just checking for `.zip` and then attemping a tar extraction.

Reply:
can you provide more details?

Reply:
For example, we check for zip, and then attempt (and ask for forgiveness)
```python
if os.path.isdir(path):
    return path
    
base, ext = os.path.splitext(path)
new_dir = base
os.makedirs(new_dir, exist_ok=True)

if ext == '.zip':
    with zipfile.ZipFile(path, 'r') as zip_ref:
        zip_ref.extractall(new_dir)
else:
    try: 
        with tarfile.open(path, 'r') as tar_ref:
            tar_ref.extractall(new_dir)
    except SPECIFICEXCEPTION as e:
        raise InvalidModelPropertiesError(f'Unsupported file extension: {ext}. The path must be a directory or a compressed file. Error: {e}')
return new_dir
```